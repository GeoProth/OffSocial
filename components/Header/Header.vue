<template>
  <div class="main-header">
    <div v-if="isLoggedIn" class="menu" @click.prevent="toggleSideNav">
      <i class="fa fa-bars fa-lg" />
    </div>
    <div class="branding" @click="() => isLoggedIn ? $router.push('/info') : $router.push('/')">
      <h3>Delete My Digital Life</h3>
    </div>
    <div v-if="isLoggedIn" class="user">
      <i v-popover:logoutButton.bottom class="fa fa-user-circle-o fa-lg" />
      <popover name="logoutButton" :width="70">
        <div class="logout-popover" @click="logout">
          <p>Log out</p>
        </div>
      </popover>
    </div>
  </div>
</template>
<script>
import { mapActions, mapGetters } from "vuex";
export default {
  name: "Header",
  computed: {
    ...mapGetters("user", ["isLoggedIn"])
  },
  methods: {
    ...mapActions("layout", ["toggleSideNav"]),
    ...mapActions("user", ["logout"])
  }
};
</script>
<style lang="stylus" scoped>
.main-header {
  display: flex
  width: 100%;
  height: 3.5rem;
  flex-wrap: nowrap;
  align-items: center;
  box-shadow: 2px -2px 15px;
}
.menu {
  order: 1;
  margin-left: 1.5rem;
  i {
    vertical-align: -5%;
  }
}
.branding {
  order: 2;
  flex-grow: 3;
  text-align: center;
}
.user {
  order: 3;
  margin-right: 1.5rem;
  i {
    vertical-align: -5%;
  }
}
.logout-popover {
  height: 100%;
  width: 100%;
  text-align: center;
  color: ui_black;
  p {
    font-size: 1rem;
  }
}
</style>
