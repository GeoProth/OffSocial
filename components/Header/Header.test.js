import Header from "./Header";

describe("Header component", () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallowMock(Header, {
      store: {
        layout: {
          actions: {
            toggleSideNav: jest.fn()
          }
        },
        user: {
          getters: {
            isLoggedIn: jest.fn()
          }
        }
      }
    });
  });
  test("Should render", () => {
    expect(wrapper.html()).toMatchSnapshot();
  });
});
