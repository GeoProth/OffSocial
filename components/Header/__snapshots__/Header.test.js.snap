// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component Should render 1`] = `
<div class="main-header">
  <!---->
  <div class="branding">
    <h3>Delete My Digital Life</h3>
  </div>
  <!---->
</div>
`;
